tests/conftest.py

```diff
@@ -995,3 +1000,13 @@ async def have_connections() -> bool:
 @pytest.fixture(name="database_uri")
 def database_uri_fixture() -> str:
     return f"file:db_{random.randint(0, 99999999)}?mode=memory&cache=shared"
+
+
+@pytest.fixture(name="available_listen_port", scope="session")
+def available_listen_port() -> int:
```

Comment:
The idea with the explicit name specification is so that the function can have a different name.  This makes it harder to accidentally reference the fixture function without having depended on it as a fixture.  This is generally just an issue within the same file.

```suggestion
def available_listen_port_fixture() -> int:
```

Reply:
ah makes sense.